docs/plugins/plugin-writer/developer-setup.rst

```diff
@@ -0,0 +1,110 @@
+      VAGRANT_SYNCED_FOLDERS.each do |host_path, guest_path|
+          # Use SSHFS instead of NFS. Requires the vagrant-sshfs plugin to be installed.
+          # Use SSHFS to share directories. The ``-o nonempty`` option is passed to allow
+          # mounts on non-empty directories.
+          # pulp3_dev.vm.synced_folder host_path, guest_path, type: "sshfs", sshfs_opts_append: "-o nonempty"
+
+          # Comment this out if you use a different filesystem (like sshfs) or while using Virtualbox provider
+            pulp3_dev.vm.synced_folder host_path, guest_path, type: "nfs", nfs_version: 4, nfs_udp: false
+      end
+
+      pulp3_dev.vm.provider :libvirt do |domain, override|
+          domain.cpus = 4
+          # In some cases, the guest gets stuck at "Waiting for domain to get an IP address..." if
+          # the default cpu_mode, 'host-model', is used. Using 'host-passthrough' cpu_mode prevents
+          # VM migration between hosts with different CPUs. However, development environments are
+          # expected to live on a single host for the duration of their existence. Due to this known
+          # issue and our use case, the default cpu_mode is overridden.
+          domain.cpu_mode = "host-passthrough"
+          domain.graphics_type = "spice"
+          domain.memory = 2048
+          domain.video_type = "qxl"
+
+          # Uncomment this to expand the disk to the given size, in GB (default is usually 40)
+          # You'll also need to uncomment the provisioning step below that resizes the root partition
+          # do not set this to a size smaller than the base box, or you will be very sad
+          # domain.machine_virtual_size = 80
+
+          # Uncomment the following line if you would like to enable libvirt's unsafe cache
+          # mode. It is called unsafe for a reason, as it causes the virtual host to ignore all
+          # fsync() calls from the guest. Only do this if you are comfortable with the possibility of
+          # your development guest becoming corrupted (in which case you should only need to do a
+          # vagrant destroy and vagrant up to get a new one).
+          #
+          # domain.volume_cache = "unsafe"
+      end
+
+      pulp3_dev.vm.provider :virtualbox do |vb, override|
+          vb.cpus = 4
+          vb.memory = 2048
+
+          # Uncomment this to expand the disk to the given size, in GB (default is usually 40)
+          # You'll also need to uncomment the provisioning step below that resizes the root partition
+          # do not set this to a size smaller than the base box, or you will be very sad
+          # vb.machine_virtual_size = 80
+      end
+    end
+  end
+
+Start up the ``pulpcore`` Vagrant box::
+
+  $ vagrant up
+
+Once the box is downloaded and booted an ssh connection can be established with a forward tunnel for port 8000::
+
+  $ vagrant ssh -- -L 8000:localhost:8000
+
+Become user ``pulp``, activate the ``pulpenv`` virtual env, and start the webserver::
```

Comment:
the log-in part is missing, I guess:

```rest
Log-in, become user ``pulp``,...

  $ vagrant ssh
  (inside the vagrant box)$ sudo su - pulp 
  ...

```

Reply:
The user is establishing an ssh connection in the previous step. So, the log-in part is already covered.